flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -486,6 +487,26 @@ public static String tryToGetPrettyPrintYaml(KubernetesResource kubernetesResour
         }
     }
 
+    /** Checks if hostNetwork is enabled. */
+    public static boolean isHostNetwork(Configuration configuration) {
+        return configuration.getBoolean(KubernetesConfigOptions.KUBERNETES_HOSTNETWORK_ENABLED);
+    }
+
+    /** Parse port from webUrl. */
+    public static Integer gerRestBindPort(String webUrl) {
```

Comment:
I believe this is duplicated with `YarnResourceManagerDriver#registerApplicationMaster`. Could you please deduplicate this by introducing a new method.

After then, this change could be splitted into a separate `hotfix` commit before all the concrete changes. Adding some corresponding tests will be great.

https://flink.apache.org/contributing/code-style-and-quality-pull-requests.html#commit-naming-conventions

Reply:
Create a new class like `ResourceManagerUtils` or add this method in `AbstractResourceManagerDriver`?

Reply:
Having the `ResourceManagerUtils` makes more senses to me.